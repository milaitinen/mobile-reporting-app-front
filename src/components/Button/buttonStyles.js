import EStyleSheet from 'react-native-extended-stylesheet';

const styles = EStyleSheet.create({
    view: {
        backgroundColor: 'transparent',
        borderRadius: '$buttonBorderRadius',
        justifyContent: 'center',
        alignContent: 'center',
        height: 70,
        marginHorizontal: 10,
        marginBottom: 2,
    },

    sendButtonContainer: {
        backgroundColor: '#99d9ad',
        borderRadius: '$buttonBorderRadius',
        borderWidth: 1,
        alignContent: 'center',
        justifyContent: 'center',
        paddingLeft: 15,
        paddingRight: 15,
        borderColor: '#99d9ad',
        '@media (max-width: 350)': {
            borderRadius: '$buttonBorderRadiusSmall',
            height: 30,
        },
        '@media (min-width: 350)': {
            height: 40,
        },
        elevation: 3,
        shadowOffset: {
            width: 0,
            height: 3,
        },
    },

    saveButtonContainer: {
        backgroundColor: '$draftBlue',
        borderRadius: '$buttonBorderRadius',
        borderWidth: 1,
        alignContent: 'center',
        justifyContent: 'center',
        paddingLeft: 15,
        paddingRight: 15,
        borderColor: '$draftBlue',
        '@media (max-width: 350)': {
            borderRadius: '$buttonBorderRadiusSmall',
            height: 30,
        },
        '@media (min-width: 350)': {
            height: 40,
        },
        elevation: 3,
        shadowOffset: {
            width: 0,
            height: 3,
        },
    },

    buttonContent: {
        flexDirection: 'row',
        justifyContent: 'center',
        alignContent: 'center',
    },

    text: {
        paddingLeft: 10,
        color: '#474c52',
        fontFamily: '$primaryFont',
        textAlign: 'center',
        '@media (max-width: 350)': {
            fontSize: 12,
        },
        '@media (min-width: 350)': {
            fontSize: 16,
        },
    },
});

export default styles;