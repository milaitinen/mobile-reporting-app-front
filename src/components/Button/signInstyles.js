import EStyleSheet from 'react-native-extended-stylesheet';

const styles = EStyleSheet.create({
    view: {
        flex: 0.7,
        flexDirection: 'row',
        justifyContent: 'center',
    },

    buttonContainer: {
        backgroundColor: '#9dcbe5',
        borderRadius: '$buttonBorderRadius',
        borderWidth: 1,
        alignContent: 'center',
        justifyContent: 'center',
        paddingLeft: 15,
        paddingRight: 15,
        borderColor: '#9dcbe5',
        '@media (max-width: 350)': {
            borderRadius: '$buttonBorderRadiusSmall',
            marginTop: 20,
            height: 30,
        },
        '@media (min-width: 350)': {
            marginTop: 30,
            marginBottom: 60,
            height: 40,
        }
    },
    text: {
        color: '#274752',
        fontFamily: '$primaryFont',
        textAlign: 'center',
        '@media (max-width: 350)': {
            fontSize: 12,
        },
        '@media (min-width: 350)': {
            fontSize: 16,
        }
    },
});

export default styles;