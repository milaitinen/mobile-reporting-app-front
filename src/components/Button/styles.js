import EStyleSheet from 'react-native-extended-stylesheet';

const styles = EStyleSheet.create({

    buttonContainer: {
        backgroundColor: '#9dcbe5',
        marginTop:30,
        paddingTop:10,
        paddingBottom:10,
        marginLeft:115,
        marginRight:115,
        marginBottom: 60,
        borderRadius:10,
        borderWidth: 1,
        borderColor: '#9dcbe5',
    },
    text: {
        color: '#274752',
        fontFamily: '$primaryFont',
        fontSize: 15,
        textAlign: 'center',
    },
});

export default styles;