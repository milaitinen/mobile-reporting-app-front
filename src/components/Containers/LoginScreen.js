import React from 'react';
import { TouchableOpacity, StyleSheet, Text, View, TextInput, ImageBackground, Image } from 'react-native';
import Icon from 'react-native-vector-icons/Feather';
import LinearGradient from 'react-native-linear-gradient';

export default class LoginScreen extends React.Component {

    constructor(props)
    {
        super(props);
        this.state = {
            isLoading: true,
            TextInputUser: '',
            TextInputPassword: '',
            TextInputServer: ''
        };
    }


    render() {
        return (
            <LinearGradient
                colors={['#3d4f7c', '#31456f', '#1b3055']}
                style={styles.contentContainer}>

                <Text
                    style={styles.title}>
                    MR-Application
                </Text>

                <Text
                    style={styles.slogan}>
                    Keep calm and keep reporting
                </Text>
                <View style={styles.SectionStyle}>
                    <Icon name={'user'} color={'white'} size={16} />
                    <TextInput
                        placeholder='Email'
                        placeholderTextColor='white'
                        onChangeText={TextInputUser => this.setState({TextInputUser})}
                        underlineColorAndroid='transparent'
                        style={styles.TextInputStyleClass}
                    />
                </View>

                <View style={styles.SectionStyle}>
                    <Icon name={'lock'} color={'white'} size={16}/>
                    <TextInput
                        secureTextEntry={true}
                        placeholder='Password'
                        placeholderTextColor='white'
                        onChangeText={TextInputPassword => this.setState({TextInputPassword})}
                        underlineColorAndroid='transparent'
                        style={styles.TextInputStyleClass}
                    />
                </View>
                <View style={styles.SectionStyle}>
                    <Icon name={'globe'} color={'white'} size={16}/>
                    <TextInput
                        placeholder='Server url'
                        placeholderTextColor='white'
                        onChangeText={TextInputServer => this.setState({TextInputServer})}
                        underlineColorAndroid='transparent'
                        style={styles.TextInputStyleClass}
                    />
                </View>

                <TouchableOpacity
                    onPress={() => this.props.navigation.navigate('drawerStack')}
                    style={styles.button}>
                    <Text style={styles.signin}>
                        Sign In
                    </Text>
                </TouchableOpacity>

                <Text
                    style={styles.textLink}
                    onPress={() => this.props.navigation.navigate('forgottenPasswordScreen')} >
                    Forgot password?
                </Text>

                <Text
                    style={styles.signup}
                    onPress={() => this.props.navigation.navigate('signupScreen')} >
                    Sign up
                </Text>

                <Text
                    style={styles.copyright}>
                    Copyright Â© Arter Oy 2017
                </Text>
            </LinearGradient>
        );
    }
}

const styles = StyleSheet.create({
    SectionStyle: {
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: 'transparent',
        borderBottomWidth: 1,
        borderBottomColor: 'white',
        alignSelf: 'center',
        height: 40,
        width: 250,
        margin: 10
    },

    TextInputStyleClass: {

        // textAlign: 'center',
        width: 215,
        height: 40,
        fontFamily: 'Roboto-Light',
        fontSize: 16,
        color: 'white',
        alignSelf: 'center',
        marginLeft: 5,
        // borderWidth: 1,
        // borderColor: '#FF5722',

        // Set border Radius.
        // borderRadius: 10 ,
    },

    button: {
        backgroundColor: '#9dcbe5',
        marginTop:30,
        paddingTop:10,
        paddingBottom:10,
        marginLeft:115,
        marginRight:115,
        marginBottom: 25,
        borderRadius:10,
        borderWidth: 1,
        borderColor: '#9dcbe5',

    },

    signin: {
        color: '#274752',
        fontFamily: 'Roboto-Light',
        fontSize: 15,
        textAlign: 'center',
    },

    contentContainer: {
        padding: 20,
        flex: 1,
        width: null,
        height: null,
        backgroundColor: 'transparent',
        justifyContent: 'center',
    },

    textLink: {
        color: 'white',
        paddingTop: 20,
        fontFamily: 'Roboto-Light',
        fontSize: 14,
        textAlign: 'center'
    },

    signup: {
        color: '#78d3f2',
        paddingTop: 25,
        fontFamily: 'Roboto-Light',
        fontSize: 15,
        textAlign: 'center'
    },

    title: {
        fontFamily: 'Roboto-Light',
        fontSize: 30,
        textAlign: 'center',
        marginTop: 10,
        paddingBottom: 30,
        color: 'white',
        // fontWeight: 'bold'
    },

    slogan: {
        fontSize: 30,
        width: 200,
        textAlign: 'center',
        paddingBottom: 30,
        color: 'white',
        fontFamily: 'Qwigley-Regular',
        alignSelf: 'center',
        // fontWeight: 'bold'
    },

    copyright: {
        fontFamily: 'Roboto-Light',
        color: 'white',
        textAlign: 'center',

        marginTop: 50,
    }
});