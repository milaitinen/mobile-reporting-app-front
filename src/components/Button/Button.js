import React from 'react';
import { View, Text, TouchableHighlight } from 'react-native';
import color from 'color';
import { Icon } from 'react-native-elements';
import styles from './buttonStyles';

// TODO: styling
const CustomButton = ({ title, type, onPress }) => {

    if (type === 'send') {
        return (
            <View style={styles.view}>
                <TouchableHighlight
                    underlayColor={color('#9dcbe5').darken(0.1)}
                    style={styles.sendButtonContainer}
                    onPress={onPress}>
                    <View style={styles.buttonContent}>
                        <Icon name={'send'} color={'#474c52'} size={18}/>
                        <Text style={styles.text}>{title}</Text>
                    </View>
                </TouchableHighlight>
            </View>
        );
    }
    return (
        <View style={styles.view}>
            <TouchableHighlight
                underlayColor={color('#8cc9e5').darken(0.1)}
                style={styles.saveButtonContainer}
                onPress={onPress}>
                <View style={styles.buttonContent}>
                    <Icon name={'save'} color={'#474c52'} size={18}/>
                    <Text style={styles.text}>{title}</Text>
                </View>
            </TouchableHighlight>
        </View>
    );
};

export default CustomButton;