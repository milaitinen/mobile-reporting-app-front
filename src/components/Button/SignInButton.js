import React from 'react';
import { Text, TouchableHighlight } from 'react-native';

import styles from './styles';

const SignInButton = ({ onPress, children }) => (
    <TouchableHighlight
        onPress={onPress}
        style={styles.touchable}
    >
        <Text style={styles.text}>{children}</Text>
    </TouchableHighlight>
);

export default SignInButton;