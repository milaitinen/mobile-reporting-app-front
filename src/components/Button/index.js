import SignInButton from './SignInButton';
import Button from './Button';
import styles from './signInstyles';

export { Button, SignInButton, styles };