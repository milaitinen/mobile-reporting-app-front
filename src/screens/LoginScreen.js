import React from 'react';
import { Text, StatusBar } from 'react-native';

import loginStyles from './style/loginStyles';
import { SignInButton } from '../components/Button';
import { Input } from '../components/TextInput';
import { AppBackground } from '../components/AppBackground';

export default class LoginScreen extends React.Component {

    constructor(props)
    {
        super(props);
        this.state = {
            isLoading: true,
            TextInputUser: '',
            TextInputPassword: '',
            TextInputServer: ''
        };
    }

    render() {
        return (
            <AppBackground>
                <StatusBar backgroundColor='#3d4f7c' barStyle='light-content'/>

                <Text style={loginStyles.title}>
                    MR-Application
                </Text>

                <Text style={loginStyles.slogan}>
                    Keep calm and keep reporting
                </Text>

                <Input
                    name={'user'}
                    placeholder='Email'
                    onChangeText={TextInputUser => this.setState({ TextInputUser })}
                />
                <Input
                    name={'lock'}
                    secureTextEntry={true}
                    placeholder='Password'
                    onChangeText={TextInputPassword => this.setState({ TextInputPassword })}
                />
                <Input
                    name={'globe'}
                    placeholder='Server url'
                    onChangeText={TextInputServer => this.setState({ TextInputServer })}
                />

                <SignInButton onPress={() => this.props.navigation.navigate('drawerStack')}>
                    Sign In
                </SignInButton>

                <Text style={loginStyles.copyright}>
                    Copyright Â© Arter Oy 2017
                </Text>
            </AppBackground>
        );
    }
}