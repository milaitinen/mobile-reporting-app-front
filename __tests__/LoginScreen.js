import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import LoginScreen from '../src/screens/LoginScreen';
import { SignInButton } from '../src/components/Button';

configure({ adapter: new Adapter() });

it('renders correctly', () => {
    const tree = renderer.create(
        <LoginScreen />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

/*it('pressing the "Sign In" button works correctly', () => {
    const opensTemplates = jest.fn();
    const component = shallow(<SignInButton onPress={ opensTemplates }/>);
    component.simulate( 'press' );
    expect(opensTemplates).toHaveBeenCalled();
});*/