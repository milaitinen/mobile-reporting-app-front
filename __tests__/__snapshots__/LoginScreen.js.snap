// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  colors={
    Array [
      undefined,
      undefined,
      undefined,
    ]
  }
  style={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={undefined}
  >
    MR-Application
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={undefined}
  >
    Keep calm and keep reporting
  </Text>
  <View
    style={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 16,
          },
          undefined,
          Object {
            "fontFamily": "Feather",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      name="user"
      onChangeText={[Function]}
      placeholder="Username"
      placeholderTextColor="white"
      style={undefined}
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 16,
          },
          undefined,
          Object {
            "fontFamily": "Feather",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      name="lock"
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="white"
      secureTextEntry={true}
      style={undefined}
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 16,
          },
          undefined,
          Object {
            "fontFamily": "Feather",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      name="globe"
      onChangeText={[Function]}
      placeholder="Server url"
      placeholderTextColor="white"
      style={undefined}
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={undefined}
      >
        Sign in
      </Text>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={undefined}
  >
    Copyright © Arter Oy 2018
  </Text>
</View>
`;
